next.clayui.com/content/docs/components/drop-down.mdx

```diff
@@ -36,16 +39,81 @@ import {DropDown} from '../../../src/components/clay/DropDown';
 	Check the <a href="https://liferay.design/lexicon">Lexicon</a> <a href="https://liferay.design/lexicon/core-components/dropdowns/">Dropdown Pattern</a> for a more in-depth look at the motivations and proper usage of this component.
 </div>
 
-DropDown is delivered for composition offering flexibility for customization.
+<div class="clay-site-alert alert alert-warning">
+	<strong class="lead">Warning</strong>
+	Before you get started with this doc, read about <a href="/docs/foundations/composing.html" target="_blank">Clay's fundamentals of composition</a> to move on. The information below is assuming you have read about <a href="/docs/foundations/composing.html" target="_blank">Clay's compositional philosophy</a> and learned about the terms.
```

Comment:
That's a good point but I disagree a little with the idea that everyone can know the composition in React the vast majority maybe, but we have a lot of people who use Clay who don't know much about React or don't understand how it works. We should link our documents and make it clear what our proposal is being like, I think implicitly leaving may not be the best way to let people guess things...

I have the thought of creating a self serving documentation where people can find all the information. The idea of ​​adding this information to all documents is that many people can parachutes into each documentation and it is important that we pass this information on to people to understand our motives and eventually narrow the learning curve about using our lib.

> I slightly worry that by over communicating a compositional philosophy, we may scare people away in that they may think it is something additional they have to read through or adopt throughout their entire project.

Yeah that can be a problem, at least in the things I wrote #2585 it seems clear that we don't want to encourage but that's the way we found to better distribute Clay. I would not find it bad that the composition was to encourage actually I find the best way to be able to reuse the components in React in the application 🙂.

But we can narrow down this information in each documentation, since we have more than 30.

Reply:
I think, like you said, a single document is likely the best way forward on describing this. Particularly if we can communicate "This is how we are distributing components, not necessarily how you have to." 

Overall just wanting to avoid any sort of overhead or headache for users when getting into clay. I would much rather have people play with examples rather than have them have to read an entire doc on composition. So the more we can promote examples, I think the better off we will be.

Reply:
> I think, like you said, a single document is likely the best way forward on describing this. Particularly if we can communicate "This is how we are distributing components, not necessarily how you have to."

Yeah, I agree with you about that. I'll take care to take a look at #2585 to make sure we're getting this message across.

> Overall just wanting to avoid any sort of overhead or headache for users when getting into clay. I would much rather have people play with examples rather than have them have to read an entire doc on composition. So the more we can promote examples, I think the better off we will be.

Yes, I don't want anyone reading this every time, but it would be essential to read it once before you start working with Clay, so it's the kind of document one will visit only once or when in doubt.